include/envoy/http/filter.h

```diff
@@ -99,17 +99,22 @@ class StreamFilterCallbacks {
 
   /**
    * Returns the route for the current request. The assumption is that the implementation can do
-   * caching where applicable to avoid multiple lookups.
+   * caching where applicable to avoid multiple lookups. If a filter has modified the headers in
+   * a way that affect routing, clearRouteCache() must be called to clear the cache.
    *
    * NOTE: This breaks down a bit if the caller knows it has modified something that would affect
```

Comment:
I would remove the first part of the note since that is documented and fixed, and just leave the last sentence.